src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/ValidationContext.cs

```diff
@@ -43,20 +50,43 @@ public class ValidationContext
         /// </summary>
         public ValidationContext()
         {
+            _boolCache = new ConcurrentDictionary<string, Lazy<Task<bool>>>();
+            _indexCache = new ConcurrentDictionary<string, Lazy<Task<PackageRegistrationIndexMetadata>>>();
+            _leafCache = new ConcurrentDictionary<string, Lazy<Task<PackageRegistrationLeafMetadata>>>();
         }
 
         public ValidationContext(
-            PackageIdentity package, 
-            IEnumerable<CatalogIndexEntry> entries, 
-            IEnumerable<DeletionAuditEntry> deletionAuditEntries, 
-            CollectorHttpClient client, 
+            PackageIdentity package,
+            IEnumerable<CatalogIndexEntry> entries,
+            IEnumerable<DeletionAuditEntry> deletionAuditEntries,
+            CollectorHttpClient client,
             CancellationToken token)
+            : this()
         {
             Package = package;
             Entries = entries;
             DeletionAuditEntries = deletionAuditEntries;
             Client = client;
             CancellationToken = token;
         }
+
+        public Task<bool> GetMemoizedBoolAsync(string key, Lazy<Task<bool>> lazyTask)
```

Comment:
All three of these could have the same method name. The parameter types are different.